src/components/box/box.test.js

```diff
@@ -0,0 +1,666 @@
+        );
+
+        getDataElementByValue("box").should("have.css", "width", minWidth);
+      }
+    );
+
+    it.each([
+      [400, "400px"],
+      [800, "600px"],
+    ])(
+      "should verify when Width is set to %s that Box width is not more than maxWidth %s",
+      (width, maxWidth) => {
+        CypressMountWithProviders(
+          <BoxComponent maxWidth={600} width={width} />
+        );
+
+        getDataElementByValue("box").should("have.css", "width", maxWidth);
+      }
+    );
+
+    it.each([
+      ["400px", "400px"],
+      ["800px", "600px"],
+    ])(
+      "should verify when Width is set to %s that Box width is not more than maxWidth %s",
+      (width, maxWidth) => {
+        CypressMountWithProviders(
+          <BoxComponent maxWidth={600} width={width} />
+        );
+
+        getDataElementByValue("box").should("have.css", "width", maxWidth);
+      }
+    );
+
+    it.each([
+      [0.1, "135px"],
+      [0.5, "600px"],
+    ])(
+      "should verify when Width is set to %s that Box width is not more than maxWidth %s",
+      (width, maxWidth) => {
+        CypressMountWithProviders(
+          <BoxComponent maxWidth={600} width={width} />
+        );
+
+        getDataElementByValue("box").should("have.css", "width", maxWidth);
+      }
+    );
+
+    it.each([
+      [400, "600px"],
+      [800, "800px"],
+    ])(
+      "should verify when Height is set to %s that Box height is not less than minHeight %s",
+      (height, minHeight) => {
+        CypressMountWithProviders(
+          <BoxComponent minHeight={600} height={height} />
+        );
+
+        getDataElementByValue("box").should("have.css", "height", minHeight);
+      }
+    );
+
+    it.each([
+      ["400px", "600px"],
+      ["800px", "800px"],
+    ])(
+      "should verify when Height is set to %s that Box height is not less than minHeight %s",
+      (height, minHeight) => {
+        CypressMountWithProviders(
+          <BoxComponent minHeight={600} height={height} />
+        );
+
+        getDataElementByValue("box").should("have.css", "height", minHeight);
+      }
+    );
+
+    it.each([
+      [400, "400px"],
+      [800, "600px"],
+    ])(
+      "should verify when Height is set to %s that Box height is not more than maxHeight %s",
+      (height, maxHeight) => {
+        CypressMountWithProviders(
+          <BoxComponent maxHeight={600} height={height} />
+        );
+
+        getDataElementByValue("box").should("have.css", "height", maxHeight);
+      }
+    );
+
+    it.each([
+      ["400px", "400px"],
+      ["800px", "600px"],
+    ])(
+      "should verify when Height is set to %s that Box height is not more than maxHeight %s",
+      (height, maxHeight) => {
+        CypressMountWithProviders(
+          <BoxComponent maxHeight={600} height={height} />
+        );
+
+        getDataElementByValue("box").should("have.css", "height", maxHeight);
+      }
+    );
+
+    it.each(["normal", "stretch", "baseline"])(
```

Comment:
```suggestion
    it.each(["normal", "stretch", "baseline", "center", "flex-start", "flex-end"])(
```
These are a few more values I think we should test